File: languages/go/entity/key/access_control_secret.go
package key

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// AccessControlSecret represents a secret used for secret-based access control.
type AccessControlSecret struct {
	Secret   string
}

// NewAccessControlSecret creates a new AccessControlSecret instance with the provided secret.
func NewAccessControlSecret(secret string) *AccessControlSecret {
	return &AccessControlSecret{
		Secret: secret,
	}
}

func (a AccessControlSecret) ToProto() *proto.AccessControlSecret {
	return &proto.AccessControlSecret{
		Secret: a.Secret,
	}
}

File: languages/js/src/entity/key/access_control_secret.ts
import * as proto from "../../bridge/proto/bloock_keys_entities";

/**
 * Represents a secret used for secret-based access control.
 */
export class AccessControlSecret {
  secret: string;

  /**
   * Creates a new AccessControlSecret instance with the provided secret.
   * @param secret
   */
  constructor(secret: string) {
    this.secret = secret;
  }

  public toProto(): proto.AccessControlSecret {
    return proto.AccessControlSecret.fromPartial({
      secret: this.secret
    });
  }
}
